docs/state/vars.md

```diff
@@ -0,0 +1,385 @@
+
+    def increment_b(self):
+        self.counter_b += 1
+
+    @rx.cached_var
+    def last_counter_b_update(self) -> str:
+        # This is updated only when `counter_b` changes.
+        return f"{self.counter_b} at {time.strftime('%H:%M:%S')}"
+
+
+def cached_var_example():
+    return rx.vstack(
+        rx.text(f"State touched at: {CachedVarState.last_touch_time}"),
+        rx.text(f"Counter A: {CachedVarState.last_counter_a_update}"),
+        rx.text(f"Counter B: {CachedVarState.last_counter_b_update}"),
+        rx.hstack(
+            rx.button("Increment A", on_click=CachedVarState.increment_a),
+            rx.button("Increment B", on_click=CachedVarState.increment_b),
+        ),
+    )
+```
+
+```python eval
+docdemo_from(CachedVarState, component=cached_var_example)
+```
+
+## Client-storage Vars
+
+You can use the browser's local storage to persist state between sessions. 
+This allows user preferences, authentication cookies, other bits of information
+to be stored on the client and accessed from different browser tabs.
+
+A client-side storage var looks and acts like a normal `str` var, except the
+default value is either `rx.Cookie` or `rx.LocalStorage` depending on where the
+value should be stored.  The key name will be based on the var name, but this
+can be overridden by passing `name="my_custom_name"` as a keyword argument.
+
+For more information see [Browser Storage](/docs/api-reference/browser/).
+
+Try entering some values in the text boxes below and then load the page in a separate 
+tab or check the storage section of browser devtools to see the values saved in the browser. 
+
+```python exec
+class ClientStorageState(State):
+    my_cookie: str = rx.Cookie("")
+    my_local_storage: str = rx.LocalStorage("")
+    custom_cookie: str = rx.Cookie(name="CustomNamedCookie", max_age=3600)
+
+
+def client_storage_example():
+    return rx.vstack(
+        rx.hstack(rx.text("my_cookie"), rx.input(value=ClientStorageState.my_cookie, on_change=ClientStorageState.set_my_cookie)),
+        rx.hstack(rx.text("my_local_storage"), rx.input(value=ClientStorageState.my_local_storage, on_change=ClientStorageState.set_my_local_storage)),
+        rx.hstack(rx.text("custom_cookie"), rx.input(value=ClientStorageState.custom_cookie, on_change=ClientStorageState.set_custom_cookie)),
+    )
+```
+
+```python eval
+docdemo_from(ClientStorageState, component=client_storage_example)
+```
+
+## Backend-only Vars
+
+Any Var in a state class that starts with an underscore is considered backend
+only and will not be syncronized with the frontend. Data associated with a
+specific session that is not directly rendered on the frontend should be stored
+in a backend-only var to reduce network traffic and improve performance.
+
+They have the advantage that they don't need to be JSON serializable, however
+they must still be cloudpickle-able to be used with redis in prod mode. They are
```

Comment:
What does it mean to be cloudpickle-able? Do we want to describe this here or give a link?